syntax = "proto3";

package dingodb.pb.common;

option java_package = "io.dingodb.common";
option cc_generic_services = true;

message VectorScalardata {
    map<string, ScalarValue> scalar_data =
        1;  // scalar data of this vector, key: scalar key, value: scalar value
            // data.
}

message ScalarField {
    oneof data {
        bool bool_data = 1;
        int32 int_data = 2;
        int64 long_data = 3;
        float float_data = 4;
        double double_data = 5;
        string string_data = 6;
        bytes bytes_data = 7;
    }
}

enum ScalarFieldType {
    NONE = 0;
    BOOL = 1;
    INT8 = 2;
    INT16 = 3;
    INT32 = 4;
    INT64 = 5;
    FLOAT32 = 6;
    DOUBLE = 7;
    STRING = 8;
    BYTES = 9;
}

message ScalarValue {
    // data type of scalar field
    ScalarFieldType field_type = 1;
    // scalar field, only one element in most cases
    repeated ScalarField fields = 2;
}

